core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedAggFunction.java

```diff
@@ -54,7 +54,8 @@ public SqlUserDefinedAggFunction(SqlIdentifier opName,
       SqlOperandTypeInference operandTypeInference,
       SqlOperandTypeChecker operandTypeChecker, AggregateFunction function,
       boolean requiresOrder, boolean requiresOver, RelDataTypeFactory typeFactory) {
-    super(Util.last(opName.names), opName, SqlKind.OTHER_FUNCTION,
+    super(Util.last(opName.names), opName, SqlKind.getSqlKindByName(opName.names.get(0)) != null
+                    ? SqlKind.valueOf(opName.names.get(0)) : SqlKind.OTHER_FUNCTION,
```

Comment:
I just want to defined a user-defind agg function with the name SUM, for example, add a sum function of String type, however, the sql kind of this function is other function, which will be filtered when find the name of SUM,  i can't find a better solution than this, if you have any idea about this, do not hesitate to help. 
By the way, I am not very clear about your answer " without giving a single check on the actual implementation", could you explain it in detail ?